3-networks/modules/transitivity/main.tf

```diff
@@ -0,0 +1,181 @@
+      direction   = "INGRESS"
+      action      = "allow"
+      sources     = []
+      ranges = [
+        # Health checkers ranges
+        "35.191.0.0/16", "130.211.0.0/22", "209.85.152.0/22", "209.85.204.0/22"
+      ]
+      targets              = [module.service_account.emails["transitivity-gw"]]
+      use_service_accounts = true
+      rules                = [{ protocol = "tcp", ports = [22] }]
+      extra_attributes     = {}
+    },
+    # Note: Actual allowed traffic is controlled via iptables within the guest environment.
+    allow-all-internal-ingress = {
+      description          = "Allow all private traffic to transitivity gateways"
+      direction            = "INGRESS"
+      action               = "allow"
+      sources              = []
+      ranges               = flatten(values(var.regional_aggregates))
+      targets              = [module.service_account.emails["transitivity-gw"]]
+      use_service_accounts = true
+      rules                = [{ protocol = "all", ports = null }]
+      extra_attributes     = {}
+    }
+    allow-all-internal-egress = {
+      description          = "Allow all private traffic to transitivity gateways"
+      direction            = "EGRESS"
+      action               = "allow"
+      sources              = []
+      ranges               = flatten(values(var.regional_aggregates))
+      targets              = [module.service_account.emails["transitivity-gw"]]
+      use_service_accounts = true
+      rules                = [{ protocol = "all", ports = null }]
+      extra_attributes     = {}
+    }
+  }
+}
+
+module "ilb_addresses" {
+  source     = "github.com/terraform-google-modules/cloud-foundation-fabric//modules/net-address?ref=v4.3.0"
+  project_id = var.project_id
+  for_each   = toset(var.regions)
+  internal_addresses = {
+    "transitivity-gw" = {
+      region     = each.key,
+      subnetwork = var.gw_subnets[each.key]
+    }
+  }
+}
+
+module "ilbs" {
+  source        = "github.com/terraform-google-modules/cloud-foundation-fabric//modules//net-ilb?ref=v4.3.0"
+  for_each      = toset(var.regions)
+  project_id    = var.project_id
+  region        = each.key
+  name          = "ilb-${each.key}"
+  network       = var.vpc_name
+  subnetwork    = var.gw_subnets[each.key]
+  address       = module.ilb_addresses[each.key].internal_addresses["transitivity-gw"].address
+  ports         = null
+  global_access = true
+  backend_config = {
+    session_affinity                = "CLIENT_IP"
+    timeout_sec                     = null
+    connection_draining_timeout_sec = null
+  }
+  backends = [{
+    failover       = false
+    group          = module.migs[each.key].instance_group
+    balancing_mode = "CONNECTION"
+    }
+  ]
+  health_check_config = {
+    type = "tcp", check = { port = 22 }, config = {}, logging = true
+  }
+}
+
+module "vpc_routes" {
+  source     = "github.com/terraform-google-modules/cloud-foundation-fabric//modules///net-vpc?ref=v4.3.0"
```

Comment:
Can we make use of the routes module instead as it is published to the registry?
https://github.com/terraform-google-modules/terraform-google-network/tree/master/modules/routes

Reply:
we could also use https://github.com/terraform-google-modules/terraform-google-lb-internal which is in the registry

Reply:
I didn't find a way to set ILB as next hop with the registry's routes module.
For the internal load balancer module, it requires me to specify ports, despite it being an all_ports = true LB.

Reply:
Created  https://github.com/terraform-google-modules/terraform-google-lb-internal/issues/45 for the issue with firewalls in the lb-internal module.

Reply:
Created https://github.com/terraform-google-modules/terraform-google-network/issues/240 for the issue with `next_hop_ilb` routes.